### DISCUSSION TITLE
No automatic validate() in Data class with custom fromRequest() method

### DISCUSSION THREAD

**Discussion by slider23** [spatie/laravel-data#28] [Q&A]
Hello. 

If the Data class has no `fromRequest()` method, then validation is called during `SomeData::from($request)`.

If the Data class has own `fromRequest()` method, then validation is not called during `SomeData::from($request)`, you must do like this:
```
public static function fromRequest(Request $request): self
    {
        self::validate($request->all()); // this
        return new self(
            // ...
        );
    }
```

Is this behavior correct ?

**Comment by rubenvanassche** [comment]
Hi @slider23,

You're right, this isn't correct behavior. It has been fixed in release 1.2.0
